internal/gatewayapi/backendtlspolicy.go

```diff
@@ -157,13 +157,54 @@ func backendTLSTargetMatched(policy gwapiv1a3.BackendTLSPolicy, target gwapiv1a2
 	return false
 }
 
-func getBackendTLSPolicy(policies []*gwapiv1a3.BackendTLSPolicy, backendRef gwapiv1a2.BackendObjectReference, backendNamespace string) *gwapiv1a3.BackendTLSPolicy {
+// getTargetBackendReferenceWithPortName returns the LocalPolicyTargetReference for the given BackendObjectReference,
+// and sets the sectionName to the port name if the BackendObjectReference is a Kubernetes Service.
+func getTargetBackendReferenceWithPortName(
+	backendRef gwapiv1a2.BackendObjectReference,
+	backendNamespace string,
+	resources *resource.Resources,
+) gwapiv1a2.LocalPolicyTargetReferenceWithSectionName {
+	ref := getTargetBackendReference(backendRef)
+	if ref.SectionName == nil {
+		return ref
+	}
+	if backendRef.Kind != nil && *backendRef.Kind != resource.KindService {
+		return ref
+	}
+
+	if service := resources.GetService(backendNamespace, string(backendRef.Name)); service != nil {
+		for _, port := range service.Spec.Ports {
+			if port.Port == int32(*backendRef.Port) {
```

Comment:
is this `backendRef.Port` a mandatory or optional field ?

Reply:
Yes, it's optional and there is a condition 

```
if ref.SectionName == nil {
		return ref
	}
```

This condition should be changed to the below to make it more understandable

```
if backendRef.Port == nil {
		return ref
	}
```